capabilities/ecr-webhook/lib/ecr.ts

```diff
@@ -0,0 +1,91 @@
+import { Log } from "pepr";
+import { getRepositoryNames } from "./utils";
+import { DeployedComponent, ZarfComponent } from "../../zarf-types";
+import { privateECRURLPattern, ECRPrivate } from "../../ecr-private";
+import { ECRPublic } from "../../ecr-public";
+import { isPrivateECRURL, isPublicECRURL } from "../../lib/utils";
+import { isECRregistry } from "../../lib/ecr";
+
+/**
+ * Creates ECR repositories for a component in the specified registry.
+ *
+ * @param {DeployedComponent} deployedComponent - The deployed Zarf component.
+ * @param {ZarfComponent} zarfComponent - The corresponding Zarf component.
+ * @param {string} registryURL - The URL of the ECR registry where repositories should be created.
+ * @returns {Promise<void>}
+ */
+export async function createRepos(
+  deployedComponent: DeployedComponent,
+  zarfComponent: ZarfComponent,
+  registryURL: string,
+): Promise<void> {
+  const result = await isECRregistry();
+
+  if (!result.isECR) {
+    throw new Error(
+      `A valid ECR URL was not found in the Zarf state secret: ${result.registryURL}\n
+      Please provide a valid ECR registry URL.\n
+      Example: '123456789012.dkr.ecr.us-east-1.amazonaws.com'`,
+    );
+  }
+
+  Log.info(
+    `Gathering a list of ECR repository names to create for component '${deployedComponent.name}'`,
+  );
+
+  const images = zarfComponent.images;
+
+  if (!images) {
+    Log.info(
+      `No repositories will be created for component '${deployedComponent.name}'`,
+    );
```

Comment:
This Log would never be seen because of line `44` in `utils.ts` in the `componentReadyForWebhook` function.

Reply:
removed